### PR TITLE
Moved ChickenBall sprite from SoupSalad.dmi to meats.dmi

### PR DESCRIPTION
**Pull request by ProgrammingGod122**
## What Does This PR Do 
This moves the sprite of ChickenBall to a more appropriate folder. I've also updated the vendor code to make sure it show's the correct icon.

## Why It's Good For The Game
Making sure that sprites are easily identifiable in the correct folder.

## Testing
I loaded the game and clicked the vendor.

## Changelog
:cl: Wartrooper
fix: Sweet and Sour Chicken is now shown in the correct place. 
/:cl:

### CODE REVIEW

**Comment by ProgrammingGod122** [comment]
This PR will also overwrite this PR
https://github.com/ParadiseSS13/Paradise/pull/23087

**Comment by DGamerL** [comment]
> This PR will also overwrite this PR #23087

So why is that PR still open then, as I assume you discussed this with them? It seems weird that we have 2 PRs that fix the same thing

**Comment by ProgrammingGod122** [comment]
> > This PR will also overwrite this PR #23087
> 
> 
> 
> So why is that PR still open then, as I assume you discussed this with them? It seems weird that we have 2 PRs that fix the same thing

I do not know that is a good question. I'll ask them.

**Comment by ProgrammingGod122** [comment]
> > This PR will also overwrite this PR #23087
> 
> 
> 
> So why is that PR still open then, as I assume you discussed this with them? It seems weird that we have 2 PRs that fix the same thing

I don't know why that PR is still open. However it doesn't fix the same thing. I was explaining it poorly this morning. My update just makes sure that food items are sorted into the correct category as well as making sure any code calling their placement is updated. That other PR is just in relation to fixing a error with a vending machine sprite.

**Comment by Henri215** [comment]
> > > This PR will also overwrite this PR #23087
> > 
> > 
> > So why is that PR still open then, as I assume you discussed this with them? It seems weird that we have 2 PRs that fix the same thing
> 
> I don't know why that PR is still open. However it doesn't fix the same thing. I was explaining it poorly this morning. My update just makes sure that this specific food item is sorted into the correct category as well as making sure any code calling their placement is updated. That other PR is just in relation to fixing a error with a vending machine sprite.

You could jusst point to Mitch that theres a better place for this sprite instead of making this one, he could very easily adapt their PR, it would still be in scope since its related to fixing that issue.

**Review by Henri215 — APPROVED** [review]
LGTM, just need to solve the merge conflict

File: code/modules/food_and_drinks/food/foods/ethnic.dm [hunk 1]
```diff
@@ -65,6 +65,7 @@
 /obj/item/reagent_containers/food/snacks/chinese/sweetsourchickenball
 	name = "sweet & sour chicken balls"
 	desc = "Is this chicken cooked? The odds are better than wok paper scissors."
+	icon_state = 'icons/obj/food/meat.dmi'
```
Comment (Henri215) [inline code/modules/food_and_drinks/food/foods/ethnic.dm:68]:
```suggestion
	icon = 'icons/obj/food/meat.dmi'
```
It is icon here, icon_state is already declared below :D

**Review by S34NW — APPROVED** [review]

**Pull request merged**